### PR TITLE
Adding Bank sync documentation

### PR DESCRIPTION
**Pull request by tjfinlinson**

### CODE REVIEW

**Comment by tjfinlinson** [comment]
This hopefully helps with the documentation item in the actual issue [#724](https://github.com/actualbudget/actual/issues/724)

**Comment by tjfinlinson** [comment]
I had some issues with my local instance. I thought I had it connected to the server, but I'm getting the error that I can't select a country. So I need to troubleshoot that a little bit so I can continue on with the steps. If anyone else that is testing wants to add steps that would be great. I also did not do any screenshots yet of anything. 

Hopefully this get's us going with something.

File: docs/Accounts/connecting-your-bank.md [hunk 1]
```diff
@@ -4,4 +4,43 @@ Title: 'Conn Bank'
...
+**Get SECRETS and KEYS for Actual**
+1. Create an account with Nordigen - https://nordigen.com/
+2. Make sure you are logged into your account dashboard page - https://ob.nordigen.com
+3. Create your user secrets by selecting user secreate from the left side menu - https://ob.nordigen.com/user-secrets/
+4. Click on the '+ create new' button at the bottom left or click the 'Create new Secrets' button the top right
+    - Make sure you download your secrets file since the **key** will not be available to you again in the account dashboard
+    - These secrets will be used in Actual to make the bank sync connection
+
+
+**Add Nordigen sync to Actual**
```
Comment (MatissJanis) [inline docs/Accounts/connecting-your-bank.md:40]: Before Nordigen can be used in the app we have one "middle" step: you need to set env variables or `config.json` for it in `actual-server`.

If using `fly.io`:
```
flyctl secrets set ACTUAL_NORDIGEN_SECRET_ID=xxxxx
flyctl secrets set ACTUAL_NORDIGEN_SECRET_KEY=xxxxx
```


For other setups I would recommend the `config.json` file:
```
{
  "nordigen": {
    "secret_id": "xxxx",
    "secret_key": "xxxxx"
  }
}
```

**Comment by tjex** [comment]
and regarding adding images, a new system is being created atm #123 . 
tl;dr: create a directory `static/img/<your-page-name>/` and save your screenshots in there.

File: docs/Accounts/connecting-your-bank.md (hunk 1 above)
Comment (fstybel) [reply docs/Accounts/connecting-your-bank.md:40]: config.json should have camelCase keys: 

```
{
  "nordigen": {
    "secretId": "xxxx",
    "secretKey": "xxxxx"
  }
}
```

Comment (Jackenmen) [reply docs/Accounts/connecting-your-bank.md:40]: When using `flyctl`, one should put both secrets in a single command so that only a single redeployment is needed:
```
flyctl secrets set ACTUAL_NORDIGEN_SECRET_ID=xxxxx ACTUAL_NORDIGEN_SECRET_KEY=xxxxx
```

**Comment by Jackenmen** [comment]
The "Connecting Your Bank" menu entry is also present in line 194 (`unavailable('Connecting Your Bank'),`), though I am not sure if it should be linked there yet.

File: docs/Accounts/connecting-your-bank.md (hunk 1 above)
Comment (tjfinlinson) [reply docs/Accounts/connecting-your-bank.md:58]: For some reason, I thought those steps were just for testing. I have added those in the documentation. I didn't know how to format the config.js section properly. Could one of you recommend the correct way?

File: docs/Accounts/connecting-your-bank.md [hunk 2]
```diff
@@ -36,6 +38,22 @@ Nordigen
     - Make sure you download your secrets file since the **key** will not be available to you again in the account dashboard
     - These secrets will be used in Actual to make the bank sync connection
 
+**Setup Config in Actual**  
+*You need to setup env variables or a config.js file before Actual can use the Nordigen integration.*
```
Comment (j-f1) [inline docs/Accounts/connecting-your-bank.md:42]: Can you link to the [dedicated config page](https://actualbudget.github.io/docs/Installing/Configuration) (`/Installing/Configuration`) and add the info about the configuration there?

Comment (tjfinlinson) [reply docs/Accounts/connecting-your-bank.md:42]: Yeah, I will move that section to the dedicated config and link to it in this page.

**Comment by tjfinlinson** [comment]
> The "Connecting Your Bank" menu entry is also present in line 194 (`unavailable('Connecting Your Bank'),`), though I am not sure if it should be linked there yet.

I saw that, and didn't know if I should turn that section on or not since this is in the experimental section in the app. So I just linked to this page from the experimental section on the docs instead. 

What do you guys think?

**Comment by j-f1** [comment]
IMO we should keep it in the experimental section but that is where the page should eventually go once the feature is stable.

File: docs/Accounts/connecting-your-bank.md (hunk 1 above)
Comment (tjfinlinson) [reply docs/Accounts/connecting-your-bank.md:46]: @Jackenmen, I made the change for the flyctl command, thank you that suggestion. Is this command run in the prompt and not put into the fly.toml file for deployment, or can you do it with either flyctl or in the toml file before deploying?

Comment (Jackenmen) [reply docs/Accounts/connecting-your-bank.md:46]: It's a command run from the terminal (such as a command prompt). `fly.toml` can't be used for setting secrets (and it's more secure since this way you don't store the secrets locally after setting them).

`flyctl secrets set` command performs a release (deploys) after setting application secrets so (assuming that you're already running an up-to-date version of Actual) there won't be a need for running the deploy command once that command finishes. Here's an example output (the command takes a minute, more or less):
```
❯  flyctl secrets set ACTUAL_NORDIGEN_SECRET_ID=xxxxx ACTUAL_NORDIGEN_SECRET_KEY=xxxxx
Release v2 created
==> Monitoring deployment
Logs: https://fly.io/apps/app-name/monitoring

 1 desired, 1 placed, 1 healthy, 0 unhealthy [health checks: 1 total, 1 passing]
--> v2 deployed successfully
```

Comment (tjfinlinson) [reply docs/Accounts/connecting-your-bank.md:46]: Sweet, thanks for the clarification. I'll make a few updates for setting the env variables in flyctl configuration section on the configuration page.

**Comment by tjfinlinson** [comment]
> and regarding adding images, a new system is being created atm #123 . tl;dr: create a directory `static/img/<your-page-name>/` and save your screenshots in there.

I will probably spin up a new instance on fly.io with the latest release to help me with getting screen shots and then I'll put them in that location you listed. Thanks for the tip. It might take me a few days to get back to this part of updating the documentation though.

**Review by MatissJanis — APPROVED** [review]
Thanks!

File: docs/Accounts/connecting-your-bank.md [hunk 3]
```diff
@@ -4,4 +4,49 @@ Title: 'Conn Bank'
...
+    - These secrets will be used in Actual to make the bank sync connection
+
+**Setup Config in Actual**  
+*You need to setup env variables or a config.js file before Actual can use the Nordigen integration.*
+
+Follow the steps provided in the [Configuring the server section](https://actualbudget.github.io/docs/Installing/Configuration)
+
+**Add Nordigen sync to Actual**
+1. Add the link to your accounts in actual (Existing or New)
+    - For an exsisting account, click on that account and select the ... (kebab menu) in the top right and choose Link Account
```
Comment (Jackenmen) [inline docs/Accounts/connecting-your-bank.md:48]: Typo + avoiding double `and` in the enumeration
```suggestion
    - For an existing account, click on that account, select the ... (kebab menu) in the top right, and choose Link Account
```

File: docs/Accounts/connecting-your-bank.md [hunk 4]
```diff
@@ -4,4 +4,49 @@ Title: 'Conn Bank'
...
+    - To create a new account with bank syncing click on the '+ Add account' link in the left menu at the bottom and select the Link bank account button
+2. Select the Link Your Bank button
+3. Select your country from the list
+4. Select your bank from the list
```
Comment (Jackenmen) [inline docs/Accounts/connecting-your-bank.md:52]: It may make sense to mention steps past that as well. i.e. after selecting the bank and clicking "Link bank in the browser", you have to go through Nordigen's access request process (logging in to the bank and granting Nordigen access to the selected accounts), and after that, you can link the accounts:




[Login to the bank and grant access]

Comment (tjfinlinson) [reply docs/Accounts/connecting-your-bank.md:52]: Thanks for the suggestions. Would that be okay for me to add these screen shots into the documentation? I haven't had success with setting up an environment with the latest version to gather some screen shots. Or if you prefer, I'm happy for you to add them in.

Comment (Jackenmen) [reply docs/Accounts/connecting-your-bank.md:52]: Feel free to use my screenshots however you want. Here's one more that might prove useful :)
image

**Review by j-f1 — APPROVED** [review]
Looks good to me! Merging so that we can more easily iterate on it.
